include/ppx/scene/scene_gltf_loader.h

```diff
@@ -0,0 +1,388 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ppx_scene_gltf_loader_h
+#define ppx_scene_gltf_loader_h
+
+#include "ppx/scene/scene_config.h"
+#include "ppx/scene/scene_loader.h"
+#include "cgltf.h"
+
+#include <set>
+
+#if defined(WIN32) && defined(LoadImage)
+#define _SAVE_MACRO_LoadImage LoadImage
+#undef LoadImage
+#endif
+
+namespace ppx {
+namespace scene {
+
+class GltfMaterialSelector
```

Comment:
Just a note: I was wondering why we needed a class for this, since it only contains a single function and makes things more complicated. But reading code below, it looks like it is so that we can override the default materials selector. Maybe worth adding a comment.